### PR TITLE
Add CookieCollection::__get()

### PR DESCRIPTION
**Pull request by andrii-pukhalevych**
PR for #17222

### CODE REVIEW

File: src/Http/Cookie/CookieCollection.php [hunk 1]
```diff
@@ -156,15 +155,20 @@ public function get(string $name): CookieInterface
      * @return bool True if the cookie exists, otherwise false.
      */
     public function has(string $name): bool
+    {
+        return null !== $this->__get($name);
+    }
+
+    public function __get(string $name): ?CookieInterface
```
Comment (markstory) [inline src/Http/Cookie/CookieCollection.php:168]: If we're going to implement `__get` we should also implement `__isset()`. Having a readable property that can't be checked with isset/empty is clunky.

**Review by markstory — COMMENTED** [review]
If your concern is primarily performance related. You could rebuild the internal storage of cookie collection so that looking up cookies by name is more efficient. I agree that the current loop based implementation isn't the most efficient.

File: src/Http/Cookie/CookieCollection.php (hunk 1 above)
Comment (andrii-pukhalevych) [reply src/Http/Cookie/CookieCollection.php:168]: @markstory Please check.

**Review by markstory — COMMENTED** [review]
Looking good, just a couple of doc strings missing.

File: tests/TestCase/Http/Cookie/CookieCollectionTest.php [hunk 2]
```diff
@@ -125,6 +125,22 @@ public function testHas(): void
         $this->assertTrue($collection->has('REMEMBER_me'), 'case insensitive cookie names');
     }
 
+    /**
+     * Tests the magic __isset() method
+     */
+    public function testMagicIsset(): void
```
Comment (markstory) [inline tests/TestCase/Http/Cookie/CookieCollectionTest.php]:
```suggestion
    public function testMagicIssetAndGet(): void
```

File: tests/TestCase/Http/Cookie/CookieCollectionTest.php [hunk 3]
```diff
@@ -125,6 +125,22 @@ public function testHas(): void
...
+    {
+        $cookies = [
+            new Cookie('remember_me', 'a'),
+            new Cookie('gtm', 'b'),
+        ];
+
+        $collection = new CookieCollection($cookies);
+        $this->assertFalse(isset($collection->nope));
+        $this->assertTrue(isset($collection->remember_me));
+        $this->assertTrue(isset($collection->REMEMBER_me));
```
Comment (markstory) [inline tests/TestCase/Http/Cookie/CookieCollectionTest.php:142]:
```suggestion
        $this->assertTrue(isset($collection->REMEMBER_me));
        $this->assertEquals('a', $collection->remember_me);
        $this->assertEquals('b', $collection->GTM);
        $this->assertNull($collection->nope);
```

**Review by markstory — COMMENTED** [review]
A few small changes to expand explicit test coverage

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**